internal/metrics/metrics.go

```diff
@@ -0,0 +1,73 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	nodeRestartReason = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "eo_elasticsearch_cr_node_restart_reason",
+			Help: "Node restarted due to Cert Restart or Rolling Restart (1 = Cert Restart, 2 = Rolling Restart, 3 = Scheduled).",
+		})
+
+	rollingRestartsCounter = prometheus.NewCounter(
```

Comment:
As per talk:
- Counters should be expressed as `****_total`
- Fold the `nodeRestartReason` metric into this metric using a label `reason={cert_restart, rolling_restart, scheduled_restart`

Consider please best practices: https://prometheus.io/docs/practices/naming/